### DISCUSSION TITLE
How to deploy contract by signing with local private key

### DISCUSSION THREAD

**Discussion by fjchen7** [ConsenSys-archive/truffle#6073] [General Q&A]
Here is my `truffle-config.js`.

```js
module.exports = {
    development: {
      network_id: "*", // Any network (default: none)
      provider: () =>
        new HDWalletProvider({
          privateKeys: ["0x9d22c..."],
          providerOrUrl: "http://127.0.0.1:8545",
        }),
    },
}
```

Here is my `1_deploy_contracts.js`

```js
const ConvertLib = artifacts.require("ConvertLib");
module.exports = async function(deployer) {
  deployer.deploy(ConvertLib);
};
```

When I run `npx truffle migrate --verbose-rpc` with the above configuration, the log prints `eth_sendTransaction`, which seems that tx is still signed by account in node rather than local private key.

I want to deploy a contract by local private key. As far as I know, I have to construct a signed transaction with my local private key, and send it by RPC `eth_sendRawTransaction` provided by node. How can I do that? Very appreciate if some could give a hint.

**Comment by FlaviuWeb3** [comment]
Hi @fjchen7 

You can find all the required steps here -> https://docs.infura.io/tutorials/ethereum/deploy-a-contract-using-web3.js
